packages/dds/tree/src/test/feature-libraries/chunked-forest/chunkEncodingEndToEnd.ts

```diff
@@ -0,0 +1,159 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { strict as assert } from "assert";
+import {
+	AllowedUpdateType,
+	ChangesetLocalId,
+	IEditableForest,
+	ITreeCursorSynchronous,
+	TreeStoredSchemaRepository,
+	mapCursorField,
+	rootFieldKey,
+} from "../../../core";
+import { leaf } from "../../../domains";
+import { typeboxValidator } from "../../../external-utilities";
+import {
+	Context,
+	DefaultChangeFamily,
+	DefaultEditBuilder,
+	FlexTreeSchema,
+	ForestSummarizer,
+	ModularChangeset,
+	TreeCompressionStrategy,
+	buildChunkedForest,
+	createMockNodeKeyManager,
+	cursorForMapTreeNode,
+	defaultSchemaPolicy,
+	getTreeContext,
+	intoStoredSchema,
+	makeFieldBatchCodec,
+	mapTreeFromCursor,
+	nodeKeyFieldKey,
+} from "../../../feature-libraries";
+// eslint-disable-next-line import/no-internal-modules
+import { TreeShape, UniformChunk } from "../../../feature-libraries/chunked-forest/uniformChunk";
+import { ForestType, SharedTreeFactory } from "../../../shared-tree";
+import { SummarizeType, TestTreeProvider, jsonSequenceRootSchema } from "../../utils";
+import { noopValidator } from "../../../codec";
+import { brand } from "../../../util";
+import {
+	Chunker,
+	defaultChunkPolicy,
+	tryShapeFromSchema,
+	// eslint-disable-next-line import/no-internal-modules
+} from "../../../feature-libraries/chunked-forest/chunkTree";
+// eslint-disable-next-line import/no-internal-modules
+import { decode } from "../../../feature-libraries/chunked-forest/codec/chunkDecoding";
+
+// TODO: Currently we split up a uniform chunk into several individual basicChunks for each node during op creation.
+// Therefore, there is currently no way for us to retrieve a uniform chunk from the tree for us to make the proper checks,
+// and the tests are expected to fail. The tests can be unskipped once uniform chunks can be inserted into the tree.
+describe.skip("End to End chunked encoding", () => {
+	it(`insert op values are correct, and shares reference with the original chunk.`, async () => {
```

Comment:
We already have decent testing for correctness of inserts, so I think it would be best to just focus on one thing, the preserved chunk identity:
```suggestion
	it(`insert ops shares reference with the original chunk`, async () => {
```

Reply:
Updated test title with suggested change